src/test/java/org/jitsi/meet/test/base/FailureListener.java

```diff
@@ -0,0 +1,362 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jitsi.meet.test.base;
+
+import org.apache.commons.io.*;
+import org.openqa.selenium.*;
+import org.openqa.selenium.logging.*;
+import org.testng.*;
+
+import java.io.*;
+import java.lang.management.*;
+import java.util.*;
+
+public class FailureListener
+    implements ITestListener
+{
+    /**
+     * The folder where the screenshost will be saved.
+     */
+    private File outputScreenshotsParentFolder = null;
+
+    /**
+     * The folder where the htmls will be saved.
+     */
+    private File outputHtmlSourceParentFolder = null;
+
+    /**
+     * The folder where the logs will be saved.
+     */
+    private static File outputLogsParentFolder = null;
+
+    @Override
+    public void onTestStart(ITestResult iTestResult)
+    {}
+
+    @Override
+    public void onTestSuccess(ITestResult iTestResult)
+    {}
+
+    @Override
+    public void onTestSkipped(ITestResult iTestResult)
+    {}
+
+    @Override
+    public void onTestFailedButWithinSuccessPercentage(ITestResult iTestResult)
+    {}
+
+    @Override
+    public void onStart(ITestContext iTestContext)
+    {
+        // default reports folder
+        outputScreenshotsParentFolder
+            = new File("target/surefire-reports/screenshots");
+
+        outputHtmlSourceParentFolder
+            = new File("target/surefire-reports/html-sources");
+        outputHtmlSourceParentFolder.mkdirs();
+
+        createLogsFolder();
+    }
+
+    @Override
+    public void onFinish(ITestContext iTestContext)
+    {}
+
+    @Override
+    public void onTestFailure(ITestResult testResult)
+    {
+        if(testResult.getThrowable() != null)
+        {
+            System.err.println("TestFailure:");
+            testResult.getThrowable().printStackTrace();
+        }
+
+        try
+        {
+            AbstractBaseTest testInstance
+                = (AbstractBaseTest)testResult.getInstance();
+
+            String fileNamePrefix
+                = testResult.getTestClass().getRealClass().getCanonicalName();
+
+            takeScreenshots(fileNamePrefix, testInstance);
+
+            saveHtmlSources(fileNamePrefix, testInstance);
+
+            saveMeetDebugLog(fileNamePrefix, testInstance);
+
+            saveBrowserLogs(fileNamePrefix, testInstance);
+
+            saveThreadDump(fileNamePrefix);
+        }
+        catch(Throwable ex)
+        {
+            ex.printStackTrace();
+        }
+    }
+
+    /**
+     * Takes screenshot of owner and participant.
+     *
+     * @param fileName the filename to use.
+     */
+    private void takeScreenshots(
+        String fileName, AbstractBaseTest testInstance)
+    {
+        takeScreenshot(testInstance.participant1.getDriver(),
+            fileName + "-owner.png");
+
+        if(testInstance.participant2 != null)
```

Comment:
no space after if - few more occurrences in this commit, probably brackets would make it easier to read since those statements are multiline

Reply:
There is a commit fixing all formatting that was introduced earlier.